### PR TITLE
[22] Newsletter Form Validation

### PR DESCRIPTION
**Pull request by faithakinbobola**
## Task: `Newsletter Form Validation`
- `I validated newsletter`

### CODE REVIEW

**Review by adaugochi — COMMENTED** [review]
Well done.

I was also expecting you to fix some of the comments I dropped on your last PR before it was merged

**Comment by faithakinbobola** [comment]
> Well done.
> 
> I was also expecting you to fix some of the comments I dropped on your last PR before it was merged

I couldn't make the changes on the PR because it was merged
I opened a new PR to fix the errors and tagged you on slack to review but one of the mentors merged it, you can check through and if there's any error to fix i'll open a new PR.

**Comment by adaugochi** [comment]
ok

File: submissions/akinbobolaFaith/dom/newsletter.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        event.preventDefault();
+    } else if (!newsletter.value.match(newsletterRegex)) {
+        newsletter.nextElementSibling.innerHTML = "Please enter a valid mail";
+        event.preventDefault();
+    } else{
+        newsletter.innerHTML = "";
+    }
+}
+
+function newsletterSignup(e) {
```
Comment (adaugochi) [inline submissions/akinbobolaFaith/dom/newsletter.js:18]: There is no need for this `newsletterSignup` function.

File: submissions/akinbobolaFaith/dom/newsletter.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    newsletterValidation(e);
+}
+
+submitForm.addEventListener("submit", newsletterSignup);
```
Comment (adaugochi) [inline submissions/akinbobolaFaith/dom/newsletter.js:22]: Simply call `newsletterValidation` function instead of `newsletterSignup`

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/akinbobolaFaith/dom/newsletter.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+let submitForm = document.querySelector('.newsletterSignup'),
+    newsletter = document.getElementById('newsletter')
+;
+
+function newsletterValidation (event) {
+    let newsletterRegex = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,6})+$/;
+    if (newsletter.value === '') {
+        newsletter.nextElementSibling.innerHTML = "Please enetr your Email Address";
```
Comment (ToluAkin) [inline submissions/akinbobolaFaith/dom/newsletter.js:8]: @faithakinbobola there is a typo in the error message

**Review by ToluAkin — CHANGES_REQUESTED** [review]

**Review by adaugochi — APPROVED** [review]

**Pull request merged**